var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -126,13 +126,20 @@ class Boost(Package):
     depends_on('mpi', when='+mpi')
     depends_on('bzip2', when='+iostreams')
     depends_on('zlib', when='+iostreams')
+    depends_on('zlib', when='%xl_r' )
```

Comment:
it's not obvious why the two dependencies are needed for xl compiler regardless of `iostreams` variant.